src/models/subscriptiondata.cpp

```diff
@@ -0,0 +1,177 @@
+  // ISO 4216. FxA passes the lowercase currency code values without
+  // transformation from Stripe.
+  m_planCurrency = planCurrency.toString().toUpper();
+
+  m_planIntervalCount = planData["interval_count"].toInt();
+  if (!m_planIntervalCount) {
+    return false;
+  }
+  logger.debug() << "Parse plan ready";
+
+  // Payment
+  QJsonObject paymentData = obj["payment"].toObject();
+
+  // If we do not receive payment data from FxA we don’t show it instead of
+  // throwing an error. There is a known bug with FxA which causes FxA to not
+  // show payment info: https://mozilla-hub.atlassian.net/browse/FXA-3856.
+  if (!paymentData.isEmpty()) {
+    // Payment provider
+    m_paymentProvider = paymentData["payment_provider"].toString();
+    // We should always get a payment provider if there is payment data
+    if (m_paymentProvider.isEmpty()) {
+      return false;
+    }
+
+    // Payment type
+    m_paymentType = paymentData["payment_type"].toString();
+
+    // For credit cards we also show card details
+    if (m_paymentType == "credit") {
+      m_creditCardBrand = paymentData["brand"].toString();
+      if (m_creditCardBrand.isEmpty()) {
+        return false;
+      }
+
+      m_creditCardLast4 = paymentData["last4"].toString();
+      if (m_creditCardLast4.isEmpty()) {
+        return false;
+      }
+
+      m_creditCardExpMonth = paymentData["exp_month"].toInt();
+      if (!m_creditCardExpMonth) {
+        return false;
+      }
+
+      m_creditCardExpYear = paymentData["exp_year"].toInt();
+      if (!m_creditCardExpYear) {
+        return false;
+      }
+    }
+  }
+
+  // Subscription
+  QJsonObject subscriptionData = obj["subscription"].toObject();
+
+  QJsonValue type = subscriptionData["_subscription_type"];
+  if (!type.isString()) {
+    return false;
+  }
+
+  // Enum from string
+  m_type = static_cast<TypeSubscription>(
+      QMetaEnum::fromType<TypeSubscription>().keyToValue(
+          type.toString().toUtf8()));
+
+  // Parse subscription data depending on subscription platform
+  switch (m_type) {
+    case web:
+      if (!parseSubscriptionDataWeb(subscriptionData)) {
+        return false;
+      }
+      break;
+    case iap_apple:
+      [[fallthrough]];
+    case iap_google:
+      if (!parseSubscriptionDataIap(subscriptionData)) {
+        return false;
+      }
+      break;
+    default:
+      logger.error() << "No matching subscription type" << type.toString();
+      return false;
+  }
+
+  m_rawJson = json;
+  emit changed();
+  logger.debug() << "Subscription data from JSON ready";
+
+  return true;
+}
+
+bool SubscriptionData::parseSubscriptionDataIap(
+    const QJsonObject& subscriptionData) {
+  logger.debug() << "Parse IAP start" << m_type;
+
+  m_expiresOn = subscriptionData["expiry_time_millis"].toInt();
+  if (!m_expiresOn) {
+    return false;
+  }
+  QJsonValue autoRenewing = subscriptionData["auto_renewing"];
+  if (!autoRenewing.isBool()) {
```

Comment:
this check can be removed:
`m_isCancelled = !subscriptionData["auto_renewing"].toBool();`